File: LifeOS/Sources/Core/Services/NLPTaskParser.swift
import Foundation
import SwiftData

class NLPTaskParser {
    static func parse(input: String) -> TaskItem {
        let title = input
        var priority = 1
        var dueDate: Date? = nil
        let lowerInput = input.lowercased()
        
        if lowerInput.contains("urgent") || lowerInput.contains("high priority") {
            priority = 2
        } else if lowerInput.contains("low priority") {
            priority = 0
        }
        
        if lowerInput.contains("tomorrow") {
            dueDate = Calendar.current.date(byAdding: .day, value: 1, to: Date())
        } else if lowerInput.contains("next week") {
            dueDate = Calendar.current.date(byAdding: .day, value: 7, to: Date())
        }
        
        return TaskItem(title: title, priority: priority, dueDate: dueDate, isCompleted: false)
    }
}

File: android/app/src/main/java/com/yashlunawat/lifeos/data/services/NLPTaskParser.kt
package com.yashlunawat.lifeos.data.services

import com.yashlunawat.lifeos.data.models.TaskItem
import java.util.Calendar

object NLPTaskParser {
    fun parse(input: String): TaskItem {
        var priority = 1
        if (input.contains("high", ignoreCase = true)) priority = 2
        if (input.contains("low", ignoreCase = true)) priority = 0
        
        var date: Long? = null
        val calendar = Calendar.getInstance()
        if (input.contains("tomorrow", ignoreCase = true)) {
            calendar.add(Calendar.DAY_OF_YEAR, 1)
            date = calendar.timeInMillis
        } else if (input.contains("today", ignoreCase = true)) {
            date = calendar.timeInMillis
        }
        
        return TaskItem(
            title = input.replace("high priority", "", ignoreCase = true)
                         .replace("low priority", "", ignoreCase = true)
                         .replace("tomorrow", "", ignoreCase = true)
                         .replace("today", "", ignoreCase = true)
                         .trim(),
            priority = priority,
            dueDate = date
        )
    }
}
